packages/server/lib/controllers/v1/logs/searchMessages.ts

```diff
@@ -0,0 +1,69 @@
+import { z } from 'zod';
+import { asyncWrapper } from '../../../utils/asyncWrapper.js';
+import type { SearchMessages } from '@nangohq/types';
+import { model, envs, operationIdRegex } from '@nangohq/logs';
+import { requireEmptyQuery, zodErrorToHTTP } from '@nangohq/utils';
+
+const validation = z
+    .object({
+        operationId: z.string().regex(operationIdRegex),
+        limit: z.number().optional().default(100),
+        search: z.string().optional(),
+        states: z
+            .array(z.enum(['all', 'waiting', 'running', 'success', 'failed', 'timeout', 'cancelled']))
+            .optional()
+            .default(['all'])
+    })
+    .strict();
+
+export const searchMessages = asyncWrapper<SearchMessages>(async (req, res) => {
+    if (!envs.NANGO_LOGS_ENABLED) {
+        res.status(404).send({ error: { code: 'feature_disabled' } });
+        return;
+    }
+
+    const emptyQuery = requireEmptyQuery(req, { withEnv: true });
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const val = validation.safeParse(req.body);
+    if (!val.success) {
+        res.status(400).send({
+            error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) }
+        });
+        return;
```

Comment:
I think using your wrapper might be the best solution because I could not come up with an abstraction that save more than 1 line (you still need to return something so you still need a if) 
but if you have something in the meantime, happy to change ☺️